src/DynamoCoreWpf/ViewModels/Core/DynamoViewModel.cs

```diff
@@ -947,8 +947,16 @@ private void Open(object parameters)
             // that can't be handled reliably
             try
             {
-                string xmlFilePath = parameters as string;
+                var xmlFilePath = parameters as string;
                 ExecuteCommand(new DynamoModel.OpenFileCommand(xmlFilePath));
+
+                if (UIDispatcher != null)
+                {
+                    // Dispatch a fit view command after all non-idle operations
+                    // are completed. This is to ensure the view's ready for it.
+                    UIDispatcher.BeginInvoke(DispatcherPriority.Background,
+                        (Action)(() => { if (CanFitView(null)) FitView(null); }));
+                }
             }
```

Comment:
Ah, the `Dispatcher` strikes again!  At some point we should move to using `SynchronizationContext`, but this is fine for now and I understand why you're doing it.  It might justify a short comment.